mlflow/models/signature.py

```diff
@@ -247,6 +254,36 @@ def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
     return ModelSignature(inputs=input_schema, outputs=output_schema)
 
 
+def _infer_signature_from_input_example(input_example, wrapped_model):
+    """
+    Infer the signature from an example input and a PyFunc wrapped model. Catches all exceptions.
+
+    :param input_example: An instance representing a typical input to the model.
+    :param wrapped_model: A PyFunc wrapped model which has a `predict` method.
+    :return: A `ModelSignature` object containing the inferred schema of both the model's inputs
+        based on the `input_example` and the model's outputs based on the prediction from the
+        `wrapped_model`.
+    """
+    try:
+        input_schema = _infer_schema(input_example)
```

Comment:
Can we slice the input example for faster deepcopy and prediction?

Reply:
We might not need this because `input_example` is (probably) small.

Reply:
Since there are so many possible data types for the `input_example` it would be complex to slice it. Happy to do so in a follow-up PR. Though, my hunch is that since users will be providing the input example themselves, I doubt that their supplied input_example will have that many records. (In fact, I feel like usually it would only have less than a handful)

Reply:
Agreed 👍